modules/programs/neovim.nix

```diff
@@ -22,6 +22,14 @@ let
     merge = mergeOneOption;
   };
 
+  extraLua51PackageType = mkOptionType {
```

Comment:
How do you anticipate handling other versions of Lua?

Reply:
Well, the current neovim that we are using in the package comes with lua5.1 hardcoded.  In future, if and when this changes, one would have to query the version of lua that neovim is built with, and call `pkgs.luaXXPackages.getLuaCPath` accordingly (where `XX` is the lua version).

Reply:
Hmm... that's kinda annoying that it's hardcoded upstream. I feel like this may introduce some pain in the future when upstream updates the Lua version. Would that break users when that happens?

Reply:
Seems unlikely that they will change it any time soon: https://www.reddit.com/r/neovim/comments/gkw86c/why_does_neovim_use_lua_51/
If they do, well, I guess that vim packages that were relying on lua libraries to be discoverable may stop working.  Note, that you can't use such packages right now.  A proper way forward would be to figure out which lua version neovim was built with and adjust the `extraLuaPackageType` accordingly.  However, that seem to be quite non-trivial task, as essentially you need to execute `nvim` binary (which we are trying to define in the derivation) and ask.
I guess all we can do right now is to write a comment of sorts.  Would you have a better idea on how to proceed?

Reply:
Ah, I see. It appears that Lua's versions don't really follow anything like semver. Considering that LuaJIT isn't available for anything besides 5.1, I have a feeling it will be a fairly large undertaking to migrate neovim away from it. Basically, as long as this isn't something that's going to happen without us noticing and then having that break users for weird and obscure reasons, I think it's fine.

I think adding a simple comment like: "Upstream Neovim is pinned on Lua 5.1 for LuaJIT support. This will need to be updated if upstream ever migrates to a newer version of Lua." would suffice. (Feel free to choose a different wording.)

Reply:
Done.